logic/system.js

```diff
@@ -11,6 +25,98 @@ async function getHiddenServiceUrl() {
     }
 };
 
+async function getAvailableUpdate() {
+    try {
+        const current = await diskLogic.readUpdateVersionFile();
+        const currentVersion = current.version;
+
+        // 'tag' should be master to begin with
+        let tag = 'ota-updates';
+        let data;
+        let isNewVersionAvailable = false;
+        let isCompatibleWithCurrentVersion = false;
+
+        // Try finding for a new update until there's a new version available
+        // which is compatible with the currently installed version
+        do {
```

Comment:
Why do while loop here over just a normal while loop?

Reply:
We loop on the condition that a new update is available and it's not compatible with the currently installed version:
https://github.com/mayankchhabra/umbrel-manager/blob/aea53add210d22d85c62d960bf80a0d7dd7ef315/logic/system.js#L59

Since we set both `isNewVersionAvailable` and `isCompatibleWithCurrentVersion` to `false`, a `do-while` ensure that the check is at least run once.

But as I type this, I don't think there's any problem in setting `isNewVersionAvailable` to `true` initially and running the standard `while` loop as it'd still execute the loop at least once.

Reply:
Updated with no issues! Thanks for pointing this out